### PR TITLE
fix: can not run blocksuite's toast function when not open a page in affine

### LINKED ISSUE #4238: Browser error after importing notion database
After importing notion database, there are some browser errors.
img_v2_cb5cd769-bbb7-45d6-93ef-3e7033a831ag
img_v2_4804f176-7323-4374-8e1a-65111f97516g
image

### PR DESCRIPTION
**Pull request by zuoxiaodong0815**
close #4238 
fix: can not run blocksuite's toast function when not open a page in affine
fix: ignore quotes when parsing table

### CODE REVIEW

**Review by doodlewind — COMMENTED** [review]
Does this have related issues to close?

**Comment by zuoxiaodong0815** [comment]
> Does this have related issues to close?

There is no corresponding issue yet. I'll create a new issue.

File: packages/blocks/src/components/import-page/import-page.ts [hunk 1]
```diff
@@ -174,16 +173,19 @@ export async function importNotion(workspace: Workspace, file: File) {
             let id = 1;
             const titles: string[] = [];
             const rows: string[][] = [];
-            tableString?.split('\n').forEach((row, index) => {
+            tableString?.split(/\r\n|\r|\n/).forEach((row, index) => {
+              const rowArray = row.split(/,\s*(?![^"]*",)/);
```
Comment (fourdim) [inline packages/blocks/src/components/import-page/import-page.ts:177]: image

Comment (zuoxiaodong0815) [reply packages/blocks/src/components/import-page/import-page.ts:177]: Separated by commas, but ignoring commas within quotes.

Comment (fourdim) [reply packages/blocks/src/components/import-page/import-page.ts:177]:
```
",", ",,", ""
   ^     ^
```
What you need to match is two, but you only match one.

Comment (zuoxiaodong0815) [reply packages/blocks/src/components/import-page/import-page.ts]: Replaced with this regular   /,\s*(?=(?:(?:[^"]*"){2})*[^"]*$)/

**Comment by zuoxiaodong0815** [comment]
> Does this have related issues to close?

I have created a new issue to relate the pr. @doodlewind

**Review by doodlewind — APPROVED** [review]

File: packages/blocks/src/components/import-page/import-page.ts (hunk 1 above)
Comment (fourdim) [reply packages/blocks/src/components/import-page/import-page.ts]: image
It's never easy parsing a string. I suggest using libraries like [papaparse](https://github.com/mholt/PapaParse).

**Pull request merged**